doc/compiler/optimizer/inlineFastPathLocations.md

```diff
@@ -0,0 +1,119 @@
+<!--
+Copyright (c) 2021, 2021 IBM Corp. and others
+
+This program and the accompanying materials are made available under
+the terms of the Eclipse Public License 2.0 which accompanies this
+distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+or the Apache License, Version 2.0 which accompanies this distribution and
+is available at https://www.apache.org/licenses/LICENSE-2.0.
+
+This Source Code may also be made available under the following
+Secondary Licenses when the conditions for such availability set
+forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+General Public License, version 2 with the GNU Classpath
+Exception [1] and GNU General Public License, version 2 with the
+OpenJDK Assembly Exception [2].
+
+[1] https://www.gnu.org/software/classpath/license.html
+[2] http://openjdk.java.net/legal/assembly-exception.html
+
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# Different Inline Fast Path Optimization Locations
+
+
+## Inline Fast Path Optimizations
+
+When there is a call to a method, the call can be executed in a cheaper way rather than making a call out to the target
+method. The inlined fast path converts a recognized call into a cheaper sequence of IL trees. One example is a call to a
+`java.lang.Math.abs()`. It can be converted into an `iabs` (absolute value of integer) [opcode](https://github.com/eclipse/omr/blob/e3a15a993c8aba80582aa1d6f3071e122acbd4c4/compiler/il/OMROpcodes.enum#L1911) .
+
+The call to `java.lang.Math.abs()` can be eliminated because there is a better representation of the execution in IL which
+is cheaper. That is a basic pattern of taking a call, recognizing them in a certain way, and optimizing it, sometimes using
+opcode, sometimes using a sequence of IL trees etc.
+
+
+There are multiple locations in our code where we do inline fast path optimizations:
+
+1. [Inliner](https://github.com/eclipse/omr/blob/master/compiler/optimizer/Inliner.cpp)
+2. [RecognizedCallTransformer](https://github.com/eclipse/omr/blob/master/compiler/optimizer/OMRRecognizedCallTransformer.cpp)
+3. [IL Generator](https://github.com/eclipse-openj9/openj9/tree/master/runtime/compiler/ilgen)
```

Comment:
It may be slightly better to have IL Generator listed first in this list since it is the earliest and simplest pass conceptually.

Reply:
I'll also move the section on ILGen to the first section.

Reply:
addressed in 190c736